Content.Shared/Roles/Jobs/SharedJobSystem.cs

```diff
@@ -76,6 +76,29 @@ public bool TryGetDepartment(string jobProto, [NotNullWhen(true)] out Department
         return false;
     }
 
+    /// <summary>
+    /// Like <see cref="TryGetDepartment"/> but ignores any non-primary departments.
+    /// For example, with CE it will return Engineering but with captain it will
+    /// not return anything, since Command is not a primary department.
+    /// </summary>
+    public bool TryGetPrimaryDepartment(string jobProto, [NotNullWhen(true)] out DepartmentPrototype? departmentPrototype)
+    {
+        // not sorting it since there should only be 1 primary department for a job.
+        var departmentProtos = _protoManager.EnumeratePrototypes<DepartmentPrototype>().ToList();
```

Comment:
Why do you need the ToList allocation

Reply:
i think IEnumerable can only be iterated once then its empty

Reply:
youre only iterating it once though

Reply:
oh i thought this was a comment on the test :trollface: